### PR TITLE
Drop Python 3.6 support

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/9981

### Does your PR introduce any breaking changes?

We no longer support Python 3.6, this follows the schedule in https://endoflife.date/python

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @carmocca @akihironitta @borda

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by akihironitta — COMMENTED** [review]
Haven't had a closer look but could this be removed as well?
https://github.com/PyTorchLightning/pytorch-lightning/blob/9826de21625b162dfef43f30864b5a6d9efab003/.circleci/config.yml#L43-L44

File: tests/conftest.py [hunk 1]
```diff
@@ -135,28 +134,8 @@ def caplog(caplog):
 
 @pytest.fixture
 def tmpdir_server(tmpdir):
-    if sys.version_info >= (3, 7):
-        Handler = partial(SimpleHTTPRequestHandler, directory=str(tmpdir))
-        from http.server import ThreadingHTTPServer
-    else:
-        # unfortunately SimpleHTTPRequestHandler doesn't accept the directory arg in python3.6
-        # so we have to hack it like this
-
```
Comment (tchaton) [inline tests/conftest.py:144]: Neat !

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by semaphore-egg** [comment]
Screenshot from 2021-12-22 21-38-08

python 3.6 is still on the support list of README.md. See left bottom of screen shot.

**Comment by akihironitta** [comment]
@semaphore-egg Thank you for pointing it out. The current release still has support for Python 3.6 and will drop it in the PL 1.6 release, and then the badge should be updated automatically, I suppose.

**Pull request merged**